File: src/test/java/io/veridia/jsonlogic/InequalityExpressionTests.java
package io.veridia.jsonlogic;

import com.fasterxml.jackson.core.JsonProcessingException;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class InequalityExpressionTests {
  private static final JsonLogic jsonLogic = new JsonLogic();

  @Test
  public void testDifferentValueSameType() throws JsonProcessingException {
    assertEquals(true, jsonLogic.apply("{\"!=\": [1, 2]}", null));
  }

  @Test
  public void testSameValueDifferentType() throws JsonProcessingException {
    assertEquals(false, jsonLogic.apply("{\"!=\": [1.0, \"1\"]}", null));
  }
}

File: src/test/scala/io/veridia/jsonlogic/InequalityExpressionTests.scala
package io.veridia.jsonlogic

import org.scalatest.funsuite.AnyFunSuite
import org.scalatest.matchers.should.Matchers

class InequalityExpressionTests extends AnyFunSuite with Matchers:
  private val jsonLogic = new JsonLogic()

  test("different value same type") {
    jsonLogic.apply("{\"!=\": [1, 2]}", null) shouldEqual true
  }

  test("same value different type") {
    jsonLogic.apply("{\"!=\": [1.0, \"1\"]}", null) shouldEqual false
  }
